skycoin/skywire
# [WIP] Skywire Android port

Did you run `make format && make check`?
Yes

Fixes #138 	

 Changes:	
- Added a separate VPN client for mobile devices
- Added `skywiremob` library to expose Skywire network API to mobile devices
- Added example Android Skywire VPN application
- Visor's startup is now fully interruptable

How to test this PR:
1. Install gomobile from https://github.com/golang/mobile
2. Install Android SDK and NDK. Put its locations to corresponding env vars: ANDROID_HOME for SDK and ANDROID_NDK_HOME for NDK
3. Compile the `skywiremob` library. To do this one should put the skywire repository according to the `GOPATH`, i.e.: `$GOPATH/src/github.com/skycoin/skywire`. Then you need to `cd` to the directory right outside of the `GOPATH`. So, having default `GOPATH` at `$HOME/go` you need to `cd $HOME`. From there you may build the library like this:
```
$ gomobile bind -o ./go/src/github.com/skycoin/skywire/cmd/skywirevisormobile/android/app/skywire.aar -target=android ./go/src/github.com/skycoin/skywire/pkg/skywiremob/
```
Probably it may be successfully built the other way, but this is the only way I got it to work. The above command will put the `skywire.aar` library right to where the Android project is looking for it.
4. Build the apk. May be done from the terminal or Android studio
5. Install and run apk
6. Put correct remote PK of visor running VPN server and passcode for that server and press `Start`. Once VPN is ready you will see the toast message
7. Check your IP address
